src/HasBinaryUuid.php

```diff
@@ -73,17 +73,90 @@ public static function decodeUuid(string $binaryUuid): string
 
     public function toArray()
     {
-        if (! $this->exists) {
-            return parent::toArray();
+        $uuidAttributes = $this->getUuidAttributes();
+
+        $array = parent::toArray();
+
+        // if (! $this->exists) {
+        // return $array;
+        // }
+
+        if (is_array($uuidAttributes)) {
+            foreach ($uuidAttributes as $attributeKey) {
+                if (array_key_exists($attributeKey, $array)) {
+                    $uuidKey = $this->getRelatedBinaryKeyName($attributeKey);
+                    $array[$attributeKey] = $this->{$uuidKey};
+                }
+            }
+        }
+
+        return $array;
+    }
+
+    public function getRelatedBinaryKeyName($attrib)
+    {
+        $suffix = $this->getUuidTextAttributeSuffix();
+
+        return "{$attrib}{$suffix}";
+    }
+
+    public function getAttribute($key)
+    {
+        if (($uuidKey = $this->uuidTextAttribute($key)) && $this->{$uuidKey} !== null) {
+            return static::decodeUuid($this->{$uuidKey});
+        }
+
+        return parent::getAttribute($key);
+    }
+
+    public function setAttribute($key, $value)
+    {
+        if ($uuidKey = $this->uuidTextAttribute($key)) {
```

Comment:
The assignment isn't required here, as `$uuidKey` isn't used?

```php
if ($this->uuidTextAttribute($key))
```

For better readability, you could even do something like this:

```php
if ($this->hasUuidTextAttribute($key))
```

`hasUuidTextAttribute` (or another name), would be a simple function returning a boolean value.